tracer/src/Datadog.Trace/Configuration/Schema/V0/MessagingSchemaV0.cs

```diff
@@ -0,0 +1,24 @@
+// <copyright file="MessagingSchemaV0.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.Configuration.Schema.V0
+{
+    internal class MessagingSchemaV0 : IMessagingSchema
```

Comment:
This is nitpicking but I believe I wouldn't have one implementation per version. I'd rather have one `MessagingSchema` class with a switch in each method. I believe it would better show the difference between schemas and wouldn't make it more complex to read (even if we get to 10 schemas).

Reply:
Also, one advantage of not separating by version is that you could add a `IsInboundOperationName` that would allow you to remove `ConsumeV0OperationName` and `ConsumeV1OperationName` from KafkaConstants as we would indeed need to check for both V0 and V1 values I assume.

Reply:
That's an interesting idea. I would still want the main interfaces to be as simple as possible, so I would not want to require the schema version in the API calls. Would we then call the `MessagingSchema` constructor with the schemaVersion and store it as an instance field?

Reply:
Done in 9319f38a309d8419f04e21173f1e4987641ab42e